### ISSUE TITLE
Hope to support the upcoming release of Nuxt 4

### ISSUE THREAD

**Issue by ikxin** [xibman/nuxt-zod-i18n#35]
### Description

After I enabled `compatibilityVersion: 4`, an error occurred because the `locales` directory was moved to the `./app` directory.

```
[21:48:13]  ERROR  Internal server error: Failed to resolve import "../app/node_modules/.pnpm/nuxt-zod-i18n@1.9.0_magicast@0.3.4_rollup@4.18.1/node_modules/nuxt-zod-i18n/dist/runtime/locales/en-GB.json" from ".nuxt/i18n.options.mjs". Does the file exist?
```

**Comment by xibman** [comment]
Hi @ikxin  i try to check during the next days  !

thanks for the report

**Comment by schillerenrico** [comment]
Yeah, same problem. Something inside module.mjs with the path resolve should be nuxt.options.appDir
Kinda made it work but just as a temp solution. Waiting for the update to come :)
Thanks man! Great work with the plugin 🙏

**Comment by hywax** [comment]
@schillerenrico @ikxin I did an experiment, I have this error too. It appears only when using `localeDetector`.

**Comment by xibman** [comment]
Sorry for the delay. I'm on holiday.

For sure next week I'm on it

**Comment by xibman** [comment]
Hi all,

The issue seems to be on Nuxt-I18n side see next issue: https://github.com/nuxt-modules/i18n/issues/2983
Fix already in progress : https://github.com/nuxt-modules/i18n/pull/3054/files

**Comment by xibman** [comment]
Hi all 
I think the problem is solved on nuxt i18n with this version: https://github.com/nuxt-modules/i18n/releases/tag/v8.4.0
So i can start working on my side to upgrade the plugin

**Comment by KevinLeneschmidt** [comment]
I get the error "Nuxt module should be a function: nuxt-zod-i18n" when using nuxt4

**Comment by xibman** [comment]
@KevinLeneschmidt  thanks for the report, work needed on my side to be compatible with nuxt4. 

I see that nuxt-i18n add code for nuxt 4 so I can do the same now.

I will drop a message in this thread when work done.

**Comment by hasan-ozbey** [comment]
I've fixed this issue by adding this line

```
restructureDir: './app/i18n'
```

inside of my `i18n` configuration.

**Comment by xibman** [comment]
@hasan-ozbey Thanks for the tips.

I plan to release a new version this weekend to support nuxt 4 and the new directory structure.

**Comment by Ragura** [comment]
> @hasan-ozbey Thanks for the tips.
> 
> I plan to release a new version this weekend to support nuxt 4 and the new directory structure.

Looking forward to it!

**Comment by xibman** [comment]
Sorry for the delay i was facing multiple bundle problem after lib update.
So new release today with all libs updated.

Next release i plan to comply with nuxt V4 requirements

**Comment by xibman** [comment]
Should be good with https://github.com/xibman/nuxt-zod-i18n/releases/tag/v1.11.4

@Ragura @hasan-ozbey

**Comment by Ragura** [comment]
Thanks for making this happen!
